code/datums/mutable_appearance.dm

```diff
@@ -18,14 +18,17 @@
 		if(!isnull(offset_const))
 			plane = GET_NEW_PLANE(plane, offset_const)
 		// That, or you need to pass in some non null object to reference
-		else if(!isnull(offset_spokesman))
+		else if(isatom(offset_spokesman))
 			// Note, we are ok with null turfs, that's not an error condition we'll just default to 0, the error would be
 			// Not passing ANYTHING in, key difference
 			var/turf/our_turf = get_turf(offset_spokesman)
 			plane = MUTATE_PLANE(plane, our_turf)
 		// otherwise if you're setting plane you better have the guts to back it up
 		else
 			stack_trace("No plane offset passed in as context for a non floating mutable appearance, ya done fucked up")
+	else if(!isnull(offset_spokesman) && !isatom(offset_spokesman))
+		stack_trace("Why did you pass in offset_spokesman as [offset_spokesman]?")
```

Comment:
you dont even document what offset_spokesman *is*. i was trying to use this code the other day and had no idea what on earth it was. give this an actually useful stack trace AND have proper documentation

Reply:
:( sorry